test/leo_ssec_base_tests.erl

```diff
@@ -0,0 +1,142 @@
+-module(leo_ssec_base_tests).
+
+-author("kunal.tyagi").
+
+-include_lib("eunit/include/eunit.hrl").
+
+-ifdef(TEST).
+-spec test() -> term(). %% SRSLY can we do better?
+-endif.
+
+-ifdef(EUNIT).
+
+%% @doc wrapper to iterate and check from Max to Min (inclusive)
+%% @private
+check_wrapper(Fn, MinValue, MaxValue) when MaxValue >= MinValue ->
+    Fn(MaxValue),
+    check_wrapper(Fn, MinValue, MaxValue - 1);
+check_wrapper(_Fn, _MinValue, _MaxValue) ->
+    true.
+
+-spec(run_test_() -> boolean()).
+run_test_() ->
+    {setup,
+     fun()  -> ok end,
+     fun(_) -> ok end,
+     [
+      {"test gen_salt/1 for input 1...100",
+       {timeout, timer:seconds(1),
+        fun() -> check_wrapper(fun check_salt/1, 1, 100) end}},
+      {"test gen_hash/1 for salt size 32...64, key 64",
+       {timeout, timer:seconds(1),
+        fun() -> check_wrapper(fun check_hash/1, 32, 64) end}},
+      {"test verify_key/1 for salt size 32...64, key 64",
+       {timeout, timer:seconds(1),
+        fun() -> check_wrapper(fun verify_key/1, 32, 64) end}},
+      {"test verify_ssec_algorithm/1",
+       {timeout, timer:seconds(1), fun test_verify_ssec_algorithm/0}},
+      {"test verify_ssec_key/1",
+       {timeout, timer:seconds(1), fun test_verify_ssec_key/0}}
+     ]}.
```

Comment:
Set more larger value like timer:seconds(3) which make the test on resource constrained environment like travis more stable.

Reply:
TALA 1d644ee